ambari-server/src/main/java/org/apache/ambari/server/orm/dao/RequestDAO.java

```diff
@@ -447,11 +432,55 @@ public long cleanup(TimeBasedCleanupPolicy policy) {
       affectedRows += cleanTableByIds(requestIds, "requestIds", "Request", policy.getToDateInMillis(),
               "RequestEntity.removeByRequestIds", RequestEntity.class);
 
+      return affectedRows;
     } catch (AmbariException e) {
       LOG.error("Error while looking up cluster with name: {}", policy.getClusterName(), e);
       throw new IllegalStateException(e);
     }
+  }
 
-    return affectedRows;
+  private Set<Long> findHostTaskIds(Set<Long> taskIds) {
+    final Set<Long> hostTaskIds = new HashSet<>();
+    final Set<Long> partialTaskIds = new HashSet<>();
+    taskIds.forEach(taskId -> {
+      if (partialTaskIds.size() <= BATCH_SIZE) {
+        partialTaskIds.add(taskId);
+      } else {
+        hostTaskIds.addAll(topologyLogicalTaskDAO.findHostTaskIdsByPhysicalTaskIds(partialTaskIds));
```

Comment:
I don't think these lookups executed for the last (or only) batch.  Can you please test with only a handful of records, too?

Reply:
Sure; I'll let you know the result (and double check the code before running the test; you might be right)